diff --git a/target/linux/octeon/image/Makefile b/target/linux/octeon/image/Makefile
--- a/target/linux/octeon/image/Makefile
+++ b/target/linux/octeon/image/Makefile
@@ -2,6 +2,8 @@
 TARGET_DEVICES += generic
 
 RV220W_CMDLINE:=-mtdparts=phys_mapped_flash:512k(boot-chain)ro,6m(stock-kernel)ro,16m(stock-rootfs)ro,1m(stock-data)ro,128k(legacy-env-gap)ro,8448k(vendor-tail)ro,128k(uboot-env)ro console=ttyS0,115200 loglevel=8 ignore_loglevel usbcore.nousb
+RV220W_FLASH_STAGE_CMDLINE:=-mtdparts=phys_mapped_flash:512k(boot-chain)ro,22m(openwrt-slot),1m(stock-data)ro,128k(legacy-env-gap)ro,8448k(vendor-tail)ro,128k(uboot-env)ro console=ttyS0,115200 loglevel=8 ignore_loglevel usbcore.nousb
+
 define Device/cisco_rv220w
   DEVICE_VENDOR := Cisco
   DEVICE_MODEL := RV220W
@@ -16,5 +18,20 @@
 endef
 TARGET_DEVICES += cisco_rv220w
 
+define Device/cisco_rv220w_flash_stage
+  DEVICE_VENDOR := Cisco
+  DEVICE_MODEL := RV220W
+  DEVICE_VARIANT := NOR writer
+  BOARD_NAME := cisco,rv220w
+  DEVICE_DTS := cn5010_cisco_rv220w
+  DEVICE_PACKAGES += -kmod-usb-dwc3-octeon
+  KERNEL := kernel-bin | strip-kernel | patch-cmdline | append-dtb-to-elf
+  KERNEL_DEPENDS := $$(wildcard $(DTS_DIR)/$(DEVICE_DTS).dts)
+  CMDLINE := $(RV220W_FLASH_STAGE_CMDLINE)
+  SUPPORTED_DEVICES += cisco,rv220w
+  IMAGES :=
+endef
+TARGET_DEVICES += cisco_rv220w_flash_stage
+
 ITUSROUTER_CMDLINE:=console=ttyS0,115200 root=/dev/mmcblk1p2 rootfstype=squashfs,ext4,f2fs rootwait
 define Device/itus_shield-router
